op-proposer/proposer/driver.go

```diff
@@ -237,21 +237,23 @@ func (l *L2OutputSubmitter) FetchL2OOOutput(ctx context.Context) (source.Proposa
 		return source.Proposal{}, false, nil
 	}
 
-	output, err := l.FetchOutput(ctx, nextCheckpointBlock)
+	proposal, err := l.FetchOutput(ctx, nextCheckpointBlock)
 	if err != nil {
 		return source.Proposal{}, false, fmt.Errorf("fetching output: %w", err)
 	}
 
 	// Always propose if it's part of the Finalized L2 chain. Or if allowed, if it's part of the safe L2 chain.
-	if output.SequenceNum > output.Legacy.FinalizedL2.Number && (!l.Cfg.AllowNonFinalized || output.SequenceNum > output.Legacy.SafeL2.Number) {
-		l.Log.Debug("Not proposing yet, L2 block is not ready for proposal",
-			"l2_proposal", output.SequenceNum,
-			"l2_safe", output.Legacy.SafeL2,
-			"l2_finalized", output.Legacy.FinalizedL2,
-			"allow_non_finalized", l.Cfg.AllowNonFinalized)
-		return output, false, nil
+	if !proposal.IsSuperRootProposal() {
```

Comment:
We probably should panic if we get a super root proposal here - this is the code for calling into the old `L2OutputOracle` contract.  May be time to remove support for this too....

Reply:
Added the panic as suggested.
I was tempted to remove the proposer l2oo in this pr. I'll create a ticket to do this later. It's long overdue.